_posts/2024-10-28-using-maven-wildfly-plugin-with-postgresql-ds.adoc

```diff
@@ -0,0 +1,295 @@
+
+[source,xml]
+----
+<execution>
+    <goals>
+        <goal>provision</goal>
+    </goals>
+</execution>
+----
+
+If you want to use the `script` with the `package` goal:
+
+[source,xml]
+----
+<execution>
+    <goals>
+        <goal>package</goal>
+    </goals>
+</execution>
+----
+
+Then you should configure the script like this:
+
+[source,xml]
+----
+<packaging-scripts>
+    <packaging-script>
+        <scripts>
+            <script>${configure.ds.script}</script>
+        </scripts>
+        <!-- Expressions resolved during server execution -->
+        <resolve-expressions>false</resolve-expressions>
+    </packaging-script>
+</packaging-scripts>
+----
+
+Here is an example that is showing the above configuration method:
+
+* https://github.com/wildfly/quickstart/blob/main/jaxrs-jwt/pom.xml#L177-L185[quickstart/jaxrs-jwt/pom.xml at main · wildfly/quickstart]
+
+To sum up, if you want to play with the `jberet-example` as noted in above by using a PostgreSQL back-end DB, you can use `podman`(or Docker) to start a container that runs a PostgreSQL database:
+
+[source,bash]
+----
+$ podman run -it -e POSTGRES_PASSWORD=123 -e POSTGRES_USER=batch_user -e POSTGRES_DB=batch_db -p 5432:5432 postgres
+----
+
+Then run:
+
+[source,bash]
+----
+$ mvn clean wildfly:dev -Dpostgres
+----
+
+You should be able to play with the example.
+
+Above is the introduction to the script configuration. Besides the `postgresql-driver` layer, the `wildfly-datasources-galleon-pack` also provide other layers as described in the documentfootnote:[https://github.com/wildfly-extras/wildfly-datasources-galleon-pack/blob/main/doc/postgresql/README.md]:
+
+____
+* `postgresql-default-datasource`: Provision the PostgreSQLDS non xa
+datasource and configures it as the default one. Depends on
+postgresql-datasource layer.
+* `postgresql-datasource`: Provision the PostgreSQLDS non xa
+datasource. Depends on postgresql-driver layer.
+* `postgresql-driver`: Provision the postgresql driver. This layer
+installs the JBoss Modules module org.postgresql.jdbc.
+____
+
+As the document described above, the `postgresql-datasource` and the `postgresql-default-datasource` layers define a `PostgreSQLDS` datasource in the provisioned server by default. In addition, the default `PostgreSQLDS` datasource uses some built-in parameters to connect to the underlying PostgreSQL database. Here are the parameters that must be defined:
+
+* `POSTGRESQL_DATABASE`
+* `POSTGRESQL_PASSWORD`
+* `POSTGRESQL_URL`
+* `POSTGRESQL_USER`
+
+
+In the example, it also contains a profile called `postgres-ds` that showing the usage of the `postgresql-datasource` layer:
+
+[source,xml]
+----
+<layers>
+    <layer>postgresql-datasource</layer>
+</layers>
+----
+
+In the profile it defines the following properties for the `PostgreSQLDS` datasource:
+
+[source,xml]
+----
+<env>
+    <POSTGRESQL_DATABASE>batch_db</POSTGRESQL_DATABASE>
```

Comment:
We should add:
```
<POSTGRESQL_JNDI>java:jboss/jsr352/batch_db</POSTGRESQL_JNDI>
```
here too since the profile adds that to use the JNDI name that is expected.